src/python/pants/binaries/binary_tool.py

```diff
@@ -29,6 +39,7 @@ class BinaryToolBase(Subsystem):
   platform_dependent: Optional[bool] = None
   archive_type: Optional[str] = None  # See pants.fs.archive.archive for valid string values.
   default_version: Optional[str] = None
+  default_digest: Optional[Digest] = None
```

Comment:
It's only possible to have a single-value `default_digest` for `platform_dependent = False` tools: because otherwise you will have a digest per platform.

Rather than an `Optional[Digest]`, I think that this (and the associated options) would need to be a map from platform+version(+etc?) to `Digest`.

Reply:
I agree! I can work this out!

Reply:
I've replaced this with `default_versions_and_digests`, which I think is pretty snazzy.